glean-core/android/build.gradle

```diff
@@ -175,6 +175,7 @@ dependencies {
 
     implementation "androidx.annotation:annotation:$versions.androidx_annotation"
     implementation "androidx.lifecycle:lifecycle-extensions:$versions.androidx_lifecycle_extensions"
+    implementation "androidx.lifecycle:lifecycle-common-java8:$versions.androidx_lifecycle_extensions"
```

Comment:
Mh, `java8`. Can we use that on older SDK levels? @pocmo, clues?

Reply:
This is the library where the `DefaultLifecycleObserver` class lives.  It's called `java8`, I believe, that class is provided for Java 8 users who can't use the newer annotation approaches.  But it should all be forward compatible with newer Javas and Kotlin.

Reply:
But is it backwards compatible with older versions of Java? And do we need to care?

Reply:
Good question.  I don't know.

Reply:
I assume this is fine. We have Java 8 compatibility enabled for all our AC modules:
- https://developer.android.com/studio/write/java8-support
- https://github.com/mozilla-mobile/android-components/blob/master/build.gradle#L126-L129

The Android build toolchain will perform bytecode transformations to make supported Java 8 features work on older versions of Android. However there are *some* Java 8 API features that are supported and cannot be backported and therefore are only available on API 24+.

I assume/hope that this artifact does not use any of those features. Looking at the code and configuration this does not seem to be the case. In fact the empty `DefaultLifecycleObserver` interface is the only class in it. The only thing it relies on are default interface methods and those are backwards compatible.
* https://android.googlesource.com/platform/frameworks/support/+/androidx-master-dev/lifecycle/lifecycle-common-java8
* https://android.googlesource.com/platform/frameworks/support/+/androidx-master-dev/lifecycle/lifecycle-common-java8/src/main/java/androidx/lifecycle/DefaultLifecycleObserver.java

Reply:
Thank you very much @pocmo! @mdboom , looks like we're good to go!

Reply:
Thanks for the feedback, @pocmo!  Let's still hold off on merging this until we've seen the effects of the proguard rules in isolation...